src/lib/autoLaunch/linuxAutoLaunch.ts

```diff
@@ -1,29 +1,44 @@
+import { AutoLaunchParams, ChildReturnData } from '../types'
+import * as native from '../native'
+import * as fs from "@tauri-apps/api/fs"
 
-module.exports = {
+const linuxAutoLaunch = {
 
-  //   enable({ appName, appPath, isHiddenOnLaunch }) {
-  //     const hiddenArg = isHiddenOnLaunch ? ' --hidden' : '';
+  async enable({ appName, appPath, hidden }: AutoLaunchParams): Promise<ChildReturnData> {
+    const response: ChildReturnData = { stderr: [], stdout: [] }
+    const hiddenArg = hidden ? ' --hidden' : '';
 
-  //     const data = `[Desktop Entry]
-  // Type=Application
-  // Version=1.0
-  // Name=${appName}
-  // Comment=${appName}startup script
-  // Exec=${appPath}${hiddenArg}
-  // StartupNotify=false
-  // Terminal=false`;
-
-  //     // return fileUtils.createFile({
-  //     //   data,
-  //     //   directory: this.getDirectory(),
-  //     //   filePath: this.getFilePath(appName)
-  //     // });
-  //   },
-
-  //   disable(appName) { return fileUtils.removeFile(this.getFilePath(appName)); },
-
-  //   isEnabled(appName) { return fileUtils.isEnabled(this.getFilePath(appName)); },
-
-  //   getDirectory() { return '~/.config/autostart/'; },
-  //   getFilePath(appName) { return `${this.getDirectory()}${appName}.desktop`; }
-};
+    const contents = `
+    [Desktop Entry]
+    Type=Application
+    Version=1.0
+    Name=${appName}
+    Comment=${appName}startup script
+    Exec=${appPath}${hiddenArg}
+    StartupNotify=false
+    Terminal=false
+  `
+    await fs.createDir(this.getDirectory()).catch(console.error)
+    await fs.writeFile({ contents, path: this.getFilePath(appName) })
+    response.stdout.push('success')
+    return response
+  },
+  async disable(appName: string): Promise<ChildReturnData> {
+    const response: ChildReturnData = { stderr: [], stdout: [] }
+    await fs.removeFile(this.getFilePath(appName))
+    response.stdout.push("success")
+    return response
+  },
+  async isEnabled(appName: string): Promise<boolean> {
+    try {
+      await fs.readTextFile(this.getFilePath(appName))
+      return true
+    } catch (error) {
+      console.error(error)
+      return false
+    }
+  },
+  getDirectory(): string { return '~/.config/autostart/'; },
+  getFilePath(appName: string): string { return `${this.getDirectory()}${appName}.desktop`; }
```

Comment:
I don't think these are supposed to be public API.

Reply:
consolidated in latest commit